pandas/core/internals/blocks.py

```diff
@@ -733,6 +733,13 @@ def _try_coerce_args(self, other):
                     type(self).__name__.lower().replace("Block", ""),
                 )
             )
+        if lib.is_scalar(other) and isna(other) and self.is_integer:
```

Comment:
this seems way too specific. where is the above expression (np.any(...)) and not self._can_hold_element(other)

not true?

Reply:
Because its looking at NAs in the opposite way.  The check above is for notna, whereas here we want to specifically catch np.nan

Reply:
I have no problem with the is_scalar and isna, its the is_integer which is way too specific here

Reply:
Totally agree that it _shouldn't_ be necessary.  In an earlier draft I had `not self._can_hold_na` instead of `self.is_integer`, but then BoolBlock started breaking.  I had planned to revisit this in another step, but would also be OK with trying to get this to the better `not self._can_hold_na` in this step too.

Reply:
ok, yes I think if you used ``_can_hold_na`` here would be ok with this change.

Reply:
OK I was able to make this prettier, but at the cost of making a check in Block.where uglier.  I think this is going to continue to be whack-a-mole until we can finish getting rid of the "try"s